core/libp2p/multi/multibase_codec/codecs/base16.cpp

```diff
@@ -55,7 +55,9 @@ namespace libp2p::multi::detail {
     if (encodingCaseIsUpper(string)) {
       return BaseError::NON_LOWERCASE_INPUT;
     }
-    return unhex(string);
+    auto r = unhex(string);
+    if(r) return r.value();
+    return r.error();
```

Comment:
return unhex does not work?

Reply:
No, it didn't work. Even with non explicit constructor of Buffer there is an error that result<vector> cannot be cast to result<Buffer>.